src/lsm/k_way_merge.zig

```diff
@@ -225,10 +228,10 @@ fn TestContext(comptime k_max: u32) type {
             return math.order(a, b);
         }
 
-        fn stream_peek(context: *const Self, stream_index: u32) error{ Empty, Drained }!u32 {
+        fn stream_peek(context: *const Self, stream_index: u32) error{ Again, EOF }!u32 {
             // TODO: test for Drained somehow as well
```

Comment:
Is this comment still relevant?

Reply:
It is, we still need to write a test case for consuming/refilling the streams during iteration.
I'm going to do that as part of the Scan PR.